docs/core/tools/dotnet-restore.md

```diff
@@ -49,6 +49,19 @@ For project-specific tooling, `dotnet restore` first restores the package in whi
 
 The behavior of the `dotnet restore` command is affected by some of the settings in the *Nuget.Config* file, if present. For example, setting the `globalPackagesFolder` in *NuGet.Config* places the restored NuGet packages in the specified folder. This is an alternative to specifying the `--packages` option on the `dotnet restore` command. For more information, see the [NuGet.Config reference](/nuget/schema/nuget-config-file).
 
+## Implicit `dotnet restore`
+
+Starting with .NET Core 2.0, `dotnet restore` is run implicitly if necessary when you issue the following commands:
+
+- [`dotnet new`](dotnet-new.md)
+- [`dotnet build`](dotnet-build.md)
+- [`dotnet run`](dotnet-run.md)
+- [`dotnet test`](dotnet-test.md)
+- [`dotnet publish`](dotnet-publish.md)
+- [`dotnet pack`](dotnet-pack.md)
+
+In most cases, you no longer need to explicitly issue the `dotnet restore` command. 
+
```

Comment:
I think it'd be good to call out that you pass --no-restore to any of those commands to disable implicit restore. and call out when it could be useful to use the command explicitly (Build systems or other forms of automation may still want to call dotnet restore explicitly to control the time in which the restore occurs and potential uses of the network.)

Reply:
Thanks, @mairaw -- this is very useful to point out. I've added it.